lib/src/components/NxFileUpload/NxMultiFileUpload/NxMultiFileUpload.tsx

```diff
@@ -0,0 +1,180 @@
+import { faExclamationCircle } from '@fortawesome/free-solid-svg-icons';
+import classnames from 'classnames';
+import { forEach, without } from 'ramda';
+
+import SelectedFile from '../SelectedFile';
+import NxFontAwesomeIcon from '../../NxFontAwesomeIcon/NxFontAwesomeIcon';
+import NxButton from '../../NxButton/NxButton';
+import { FormAriaContext } from '../../NxForm/context';
+import { useUniqueId } from '../../../util/idUtil';
+import { Props, propTypes, MultiSelectedFileProps } from '../types';
+
+export { Props };
+
+import '../NxFileUpload.scss';
+
+function SelectedFileWrapper({file, onDismiss: onDismissProp}: MultiSelectedFileProps) {
+  const fileRef = useRef<HTMLSpanElement>(null);
+
+  function onDismiss() {
+    onDismissProp(file);
+  }
+
+  useEffect(function() {
+    if (fileRef.current) {
+      fileRef.current.scrollIntoView({block: 'nearest'});
+    }
+  }, []);
+
+  return (
+    <SelectedFile file={file} onDismiss={onDismiss} ref={fileRef}/>
+  );
+}
+
+const NxMultiFileUpload = forwardRef<HTMLDivElement, Props>(function NxMultiFileUpload(props, ref) {
+  const {
+        className: classNameProp,
+        onChange: onChangeProp,
+        files,
+        isRequired,
+        isPristine,
+        id,
+        disabled,
+        ...attrs
+      } = props,
+      isFileSelected = !!files?.length,
+      { showValidationErrors: formShowValidationErrors } = useContext(FormAriaContext),
+      showValidationErrors = formShowValidationErrors || !isPristine,
+      showError = isRequired && showValidationErrors && !isFileSelected,
+      className = classnames('nx-multi-file-upload', classNameProp),
+      noFileMessageClassName = classnames('nx-multi-file-upload__no-file-message', {
+        'nx-multi-file-upload__no-file-message--invalid': showError
+      }),
+      inputRef = useRef<HTMLInputElement>(null),
+      inputId = useUniqueId('nx-multi-file-upload-input', id),
+      selectedFilesContainerRef = useRef<HTMLDivElement>(null),
+      validationErrorId = useUniqueId('nx-multi-file-upload-validation-error');
+
+  function onChange(evt: FormEvent<HTMLInputElement>) {
+    const dataTransferObject = new DataTransfer();
+
+    // get the previous files uploaded
+    if (files) {
+      forEach((f) => dataTransferObject.items.add(f), Array.from(files));
+    }
+
+    const inputFiles = evt.currentTarget.files,
+        normalizedFiles = !inputFiles?.length ? null : inputFiles;
+
+    if (normalizedFiles) {
+      forEach((f) => dataTransferObject.items.add(f), Array.from(normalizedFiles));
+    }
+
+    onChangeProp(dataTransferObject.files);
+  }
+
+  function openPicker() {
+    inputRef.current?.click();
+  }
+
+  function onDismiss(fileObj: File) {
+    const closeBtns = Array.from(selectedFilesContainerRef.current?.
+        querySelectorAll<HTMLButtonElement>('.nx-selected-file__dismiss-btn') ?? []);
+
+    if (files) {
+      const filesArray = Array.from(files);
+      filesArray.map((f) => {
```

Comment:
This function isn't mapping, it's executing side-effects.  Use a for loop (or at least a `forEach` function) rather than `map()`

Reply:
> This function isn't mapping, it's executing side-effects. Use a for loop (or at least a forEach function) rather than map()

Actually it looks like what you need here is [findIndex](https://ramdajs.com/docs/#findIndex)

Reply:
> Actually it looks like what you need here is [findIndex](https://ramdajs.com/docs/#findIndex)

focusing changes implemented here: [f54a520](https://github.com/sonatype/sonatype-react-shared-components/pull/850/commits/f54a52087f5c4794ea2ceff284421263f955cebd)